server/common/config/external_config.py

```diff
@@ -0,0 +1,103 @@
+import os
+
+from server.common.config.base_config import BaseConfig
+from server.common.errors import ConfigurationError
+from server.common.config import get_secret_key
+from server.common.errors import SecretKeyRetrievalError
+
+
+class ExternalConfig(BaseConfig):
+    """Manages the config attribute associated with external configuration sources, such as
+    environment variables or the AWS Secrets Manager."""
+
+    def __init__(self, app_config, default_config):
+        super().__init__(app_config, default_config)
+        try:
+            self.environment = default_config["environment"]
+            self.aws_secrets_manager__region = default_config["aws_secrets_manager"]["region"]
+            self.aws_secrets_manager__secrets = default_config["aws_secrets_manager"]["secrets"]
+
+        except KeyError as e:
+            raise ConfigurationError(f"Unexpected config: {str(e)}")
+
+    @staticmethod
+    def convert_value(value: str):
+        """convert the external value (a string), to the most appropriate type"""
+        if value == "true":
+            return True
+        if value == "false":
+            return False
+        if value == "null":
+            return None
+        try:
+            return eval(value)
+        except (NameError, SyntaxError):
+            pass
+
+        return value
+
+    def complete_config(self, context):
+        self.handle_environment(context)
+        self.handle_aws_secrets_manager(context)
+
+    def handle_environment(self, context):
```

Comment:
Add docstring please. I think this function is complex enough that a brief docstring would be helpful.